pilot/pkg/bootstrap/server_test.go

```diff
@@ -40,28 +40,69 @@ import (
 	"istio.io/pkg/filewatcher"
 )
 
-func TestNewServerCertInit(t *testing.T) {
-	configDir := t.TempDir()
+func loadCertFilesAtPaths(t TLSFSLoadPaths) error {
+	// create cert directories if not existing
+	if err := os.MkdirAll(filepath.Dir(t.testTLSCertFilePath), os.ModePerm); err != nil {
+		return fmt.Errorf("Mkdirall(%v) failed: %v", t.testTLSCertFilePath, err)
+	}
 
-	certsDir := t.TempDir()
+	if err := os.MkdirAll(filepath.Dir(t.testTLSKeyFilePath), os.ModePerm); err != nil {
+		return fmt.Errorf("Mkdirall(%v) failed: %v", t.testTLSKeyFilePath, err)
+	}
 
-	certFile := filepath.Join(certsDir, "cert-file.pem")
-	keyFile := filepath.Join(certsDir, "key-file.pem")
-	caCertFile := filepath.Join(certsDir, "ca-cert.pem")
+	if err := os.MkdirAll(filepath.Dir(t.testCaCertFilePath), os.ModePerm); err != nil {
+		return fmt.Errorf("Mkdirall(%v) failed: %v", t.testCaCertFilePath, err)
+	}
 
 	// load key and cert files.
-	if err := os.WriteFile(certFile, testcerts.ServerCert, 0o644); err != nil { // nolint: vetshadow
-		t.Fatalf("WriteFile(%v) failed: %v", certFile, err)
+	if err := os.WriteFile(t.testTLSCertFilePath, testcerts.ServerCert, 0o644); err != nil { // nolint: vetshadow
+		return fmt.Errorf("WriteFile(%v) failed: %v", t.testTLSCertFilePath, err)
 	}
-	if err := os.WriteFile(keyFile, testcerts.ServerKey, 0o644); err != nil { // nolint: vetshadow
-		t.Fatalf("WriteFile(%v) failed: %v", keyFile, err)
+	if err := os.WriteFile(t.testTLSKeyFilePath, testcerts.ServerKey, 0o644); err != nil { // nolint: vetshadow
+		return fmt.Errorf("WriteFile(%v) failed: %v", t.testTLSKeyFilePath, err)
+	}
+	if err := os.WriteFile(t.testCaCertFilePath, testcerts.CACert, 0o644); err != nil { // nolint: vetshadow
+		return fmt.Errorf("WriteFile(%v) failed: %v", t.testCaCertFilePath, err)
 	}
-	if err := os.WriteFile(caCertFile, testcerts.CACert, 0o644); err != nil { // nolint: vetshadow
-		t.Fatalf("WriteFile(%v) failed: %v", caCertFile, err)
+
+	return nil
+}
+
+func cleanupCertFileSystemFiles(t TLSFSLoadPaths) error {
+	if err := os.Remove(t.testTLSCertFilePath); err != nil {
+		return fmt.Errorf("Test cleanup failed, could not delete %s", t.testTLSCertFilePath)
 	}
 
+	if err := os.Remove(t.testTLSKeyFilePath); err != nil {
+		return fmt.Errorf("Test cleanup failed, could not delete %s", t.testTLSKeyFilePath)
+	}
+
+	if err := os.Remove(t.testCaCertFilePath); err != nil {
+		return fmt.Errorf("Test cleanup failed, could not delete %s", t.testCaCertFilePath)
+	}
+	return nil
+}
+
+// This struct will indicate for each test case
+// where tls assets will be loaded on disk
+type TLSFSLoadPaths struct {
+	testTLSCertFilePath string
+	testTLSKeyFilePath  string
+	testCaCertFilePath  string
+}
+
+func TestNewServerCertInit(t *testing.T) {
```

Comment:
this requires root to run

Reply:
ah yes i wasn't running this within my laptop but on CI instead so forgot about this, this is due to[ this other issue](https://github.com/istio/istio/issues/39303) i have to run on my laptop but that's besides the point. thanks!